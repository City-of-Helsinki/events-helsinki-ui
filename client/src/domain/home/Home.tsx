import React from "react";
import { useTranslation } from "react-i18next";

import CollectionCardContainer from "../../common/components/collectionCard/CollectionCardContainer";
import Hero from "../../common/components/hero/Hero";
import Container from "../app/layout/Container";
import Layout from "../app/layout/Layout";
import styles from "./home.module.scss";
import Search from "./Search";

const Home: React.FC = () => {
  const { t } = useTranslation();
  return (
    <Layout>
      <Hero
        buttonText={t("home.hero.buttonText")}
        subTitle={t("home.hero.subTitle")}
        title={t("home.hero.title")}
      />
      <Container>
        <Search />
      </Container>
      <div className={styles.collectionCardContainer}>
        <Container>
          <h2>{t("home.collections.title")}</h2>
          <CollectionCardContainer
            cards={[
              {
                count: 140,
                description: "Kokosimme parhaat tärpit kesälle 2019",
                id: "1",
                title: "Paras kesä ikinä"
              }
            ]}
            size="lg"
          />

          <CollectionCardContainer
            cards={[
              {
                count: 140,
                description: "",
                id: "2",
                title: "Olipa kerra ihana kesä ja tuleva syksy."
              },
              {
                count: 140,
                description: "",
                id: "3",
                title: "Olipa kerra ihana kesä ja tuleva syksy."
              }
            ]}
            size="md"
          />

          <CollectionCardContainer
            cards={[
              {
                count: 140,
                description: "",
                id: "4",
                title: "Menoa ja meininkiä viikon lopuksi"
              },
              {
                count: 140,
                description: "",
                id: "5",
                title: "Vastinetta Netflixille"
              },
              {
                count: 140,
                description: "",
                id: "6",
                title: "Menoa ja meininkiä viikon lopuksi"
              },
              {
                count: 140,
                description: "",
                id: "7",
                title: "Vastinetta Netflixille"
              }
            ]}
            size="sm"
          />
        </Container>
      </div>
    </Layout>
  );
};

export default Home;
