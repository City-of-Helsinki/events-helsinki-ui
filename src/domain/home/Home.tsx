import React from "react";
import { useTranslation } from "react-i18next";

import CollectionCardContainer from "../../common/components/collectionCard/CollectionCardContainer";
import Hero from "../../common/components/hero/Hero";
import { useCollectionListQuery } from "../../generated/graphql";
import Container from "../app/layout/Container";
import PageWrapper from "../app/layout/PageWrapper";
import styles from "./home.module.scss";
import Search from "./Search";

const Home: React.FC = () => {
  const { t } = useTranslation();

  const { data: collectionsData } = useCollectionListQuery({ ssr: false });
  const collections = collectionsData
    ? collectionsData.collectionList.data
    : [];
  const lgCollections = collections.slice(0, 1);
  const mdCollections = collections.slice(1, 3);
  const smCollections = collections.slice(3);

  return (
    <PageWrapper>
      <Hero
        buttonText={t("home.hero.buttonText")}
        subTitle={t("home.hero.subTitle")}
        title={t("home.hero.title")}
      />
      <Container>
        <Search />
      </Container>
      <div className={styles.collectionCardContainer}>
        <Container>
          <h2>{t("home.collections.title")}</h2>
          <CollectionCardContainer collections={lgCollections} size="lg" />

          <CollectionCardContainer collections={mdCollections} size="md" />

          <CollectionCardContainer collections={smCollections} size="sm" />
        </Container>
      </div>
    </PageWrapper>
  );
};

export default Home;
