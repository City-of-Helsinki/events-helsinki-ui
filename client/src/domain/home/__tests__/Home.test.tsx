import * as React from "react";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";

import Home from "../Home";

test("Home matches snapshot", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

export {};
