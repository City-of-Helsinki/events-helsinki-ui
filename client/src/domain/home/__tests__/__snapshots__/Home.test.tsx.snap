// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div
  className="pageWrapper"
>
  <header
    className="headerWrapper"
  >
    <div
      className="container"
    >
      <div
        className="navbarTop"
      >
        <div
          className="logoWrapper"
        >
          <div
            className="logo"
            onClick={[Function]}
          />
          <h1
            className="appName"
          >
            Tapahtumat
          </h1>
        </div>
        <div
          className="linkWrapper"
        >
          <a
            className="iconLink"
            href="/search"
            onClick={[Function]}
          >
            <div
              className="iconWrapper"
            >
              <svg>
                search.svg
              </svg>
            </div>
            <div
              className="textWrapper"
            >
              Etsi tekemistä
            </div>
          </a>
          <a
            className="iconLink"
            href="/search"
            onClick={[Function]}
          >
            <div
              className="iconWrapper"
            >
              <svg>
                smile.svg
              </svg>
            </div>
            <div
              className="textWrapper"
            >
              Selaa kokoelmia
            </div>
          </a>
        </div>
        <div
          className="languageDropdown"
        >
          <div
            className="languageDropdownWrapper"
          >
            <div
              className="select css-2b097c-container"
              onKeyDown={[Function]}
            >
              <div
                className="select__control css-yk16xz-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="select__value-container select__value-container--has-value css-1hwfws3"
                >
                  <div
                    className="select__single-value css-1uccc91-singleValue"
                  >
                    FI
                  </div>
                  <input
                    className="css-62g3xt-dummyInput"
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    readOnly={true}
                    tabIndex="0"
                    value=""
                  />
                </div>
                <div
                  className="select__indicators css-1hb7zxy-IndicatorsContainer"
                >
                  <div
                    className="select__dropdown-indicator"
                  >
                    <svg>
                      angle-down.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="mobileNavigation"
  >
    <header
      className="mobileHeaderWrapper"
    >
      <div
        className="container"
      >
        <div
          className="mobileNavbar"
        >
          <div
            className="logoWrapper"
          >
            <div
              className="logo"
              onClick={[Function]}
            />
            <h1
              className="appName"
            >
              Tapahtumat
            </h1>
          </div>
          <div
            className="buttonWrapper"
          >
            <button
              aria-label="Avaa valikko"
              className="menuButton"
              onClick={[Function]}
            >
              <svg>
                menu.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      className="mobileNavigationMenu"
    >
      <div
        className="menuItem searchCategories"
        onClick={[Function]}
      >
        <div
          className="text"
        >
          Selaa kategorioita
        </div>
        <div
          className="iconWrapper"
        >
          <svg>
            tags-search.svg
          </svg>
        </div>
      </div>
      <div
        className="menuItem searchEvents"
        onClick={[Function]}
      >
        <div
          className="text"
        >
          Etsi tekemistä
        </div>
        <div
          className="iconWrapper"
        >
          <svg>
            one-eye-smiley.svg
          </svg>
        </div>
      </div>
      <div
        className="menuItem mostPopular"
        onClick={[Function]}
      >
        <div
          className="text"
        >
          Suosituimmat haut
        </div>
        <div
          className="iconWrapper"
        >
          <svg>
            ranking-stars-ribbon.svg
          </svg>
        </div>
      </div>
      <div
        className="menuItem aboutService"
        onClick={[Function]}
      >
        <div
          className="text"
        >
          Tietoa palvelusta
        </div>
        <div
          className="iconWrapper"
        >
          <svg>
            flag-information.svg
          </svg>
        </div>
      </div>
      <div
        className="login"
        onClick={[Function]}
      >
        <div
          className="squareExtension"
        />
      </div>
    </div>
  </div>
  <div
    className="pageBody"
  >
    <div
      className="container"
    >
      <div
        className="searchContainer"
      >
        <button
          className="supriseMeButton"
          onClick={[Function]}
          type="button"
        >
          <div
            className="text"
          >
            Yllätä minut
          </div>
          <svg>
            suprise-me.svg
          </svg>
        </button>
        <div
          className="titleWrapper"
        >
          <h3>
            Löydä itsellesi tekemistä!
          </h3>
        </div>
        <div
          className="autosuggestWrapper"
        >
          <label>
            Mitä etsit?
          </label>
          <div
            className="searchAutosuggest"
            onClick={[Function]}
          >
            <div
              className="iconWrapper"
            >
              <svg>
                search.svg
              </svg>
            </div>
            <div />
            <div
              className="inputWrapper"
            >
              <input
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Aloita kirjoittamalla tähän, esim. teatteri"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="dateSelectorWrapper"
        >
          <label>
            Mille ajankohdalle etsit?
          </label>
          <div
            className="dateSelector"
          >
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              <div
                className="iconWrapper"
              >
                <svg>
                  calendar.svg
                </svg>
              </div>
              <div
                className="info"
              >
                Valitse ajankohta
              </div>
              <div
                className="arrowWrapper"
              >
                <svg>
                  angle-down.svg
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          className="buttonWrapper"
        >
          <button
            className="btn btnMediumPrimary btnBlock"
            onClick={[Function]}
            type="button"
          >
            <svg>
              search.svg
            </svg>
            Hae
          </button>
        </div>
      </div>
      <div
        className="categoryFilters"
      >
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              movie.svg
            </svg>
            <span>
              Elokuva
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              music.svg
            </svg>
            <span>
              Musiikki
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              sport.svg
            </svg>
            <span>
              Liikunta ja urheilu
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              museum.svg
            </svg>
            <span>
              Museot
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              dance.svg
            </svg>
            <span>
              Tanssi
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              culture.svg
            </svg>
            <span>
              Kulttuuri ja taide
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              nature.svg
            </svg>
            <span>
              Luonto ja ulkoilu
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              influence.svg
            </svg>
            <span>
              Osallistu ja vaikuta
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              theatre.svg
            </svg>
            <span>
              Teatteri
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
        <div
          className="categoryFilter"
        >
          <button
            onClick={[Function]}
          >
            <svg>
              food.svg
            </svg>
            <span>
              Ruoka
            </span>
            <svg>
              angle-right.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footerWrapper"
  >
    <div
      className="container"
    >
      <div
        className="footer"
      >
        <div
          className="helsinkiLogo"
        />
        <div
          className="copyright"
        >
          <p>
            © Copyright 2019 • All right reserved •
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
